qcodes/tests/dataset/test_measurement_context_manager.py

```diff
@@ -416,6 +416,23 @@ def test_measurement_name(experiment, DAC, DMM):
         assert datasaver.dataset.table_name == expected_name
 
 
+def test_measurement_name_param(experiment, DAC, DMM):
```

Comment:
could you also add a test where name argument is not passes and assert that the name of the dataset becomes "results"?